atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1949,16 +1961,36 @@ protected KeyValueService getKeyValueService() {
     }
 
     private Timer getTimer(String name) {
-        return metricRegistry.timer(MetricRegistry.name(SnapshotTransaction.class, name));
+        return metricsManager.registerTimer(SnapshotTransaction.class, name);
     }
 
     private Histogram getHistogram(String name) {
-        return metricRegistry.histogram(MetricRegistry.name(SnapshotTransaction.class, name));
+        return metricsManager.registerHistogram(SnapshotTransaction.class, name);
     }
 
     private Meter getTransactionConflictsMeter() {
+        return getMeter("SnapshotTransactionConflict");
+    }
+
+    private Meter getGreaterCommitTsCellFilterCount() {
+        return getMeter("commitTsGreaterThatTxTsCellFilterCount");
+    }
+
+    private Meter getInvalidStartTsTsCellFilterCount() {
+        return getMeter("invalidStartTsTsCellFilterCount");
+    }
+
+    private Meter getInvalidCommitTsCellFilterCount() {
+        return getMeter("invalidCommitTsCellFilterCount");
+    }
+
+    private static Meter getEmptyValuesCountMeter() {
+        return getMeter("emptyValuesCount");
+    }
+
+    private static Meter getMeter(String name) {
         // TODO(hsaraogi): add table names as a tag
-        return metricRegistry.meter(MetricRegistry.name(SnapshotTransaction.class, "SnapshotTransactionConflict"));
+        return metricsManager.registerMeter(SnapshotTransaction.class, name);
```

Comment:
it's a little scary that this is calling "registerMeter", maybe that method should be renamed to indicate it's idempotent.

Reply:
Done.